lib/simulateSpendCondTxLocally.js

```diff
@@ -0,0 +1,289 @@
+        if (ii !== -1) {
+          outsGroupedByColorFrom[color][out.from][ii] = { to: out.to, amount: add(out.amount, outsGroupedByColorFrom[color][out.from][ii].amount) }
+        } else{
+          outsGroupedByColorFrom[color][out.from].push({ to: out.to, amount: out.amount })
+        }
+      } else {
+        outsGroupedByColorFrom[color][out.from] = [{ to: out.to, amount: out.amount }]
+      }
+    }
+    )
+  })
+  return outsGroupedByColorFrom
+}
+
+function calcOutputs(unspent, inputs, aggregatedOuts) {
+  if (unspent.length === 0) {
+    throw new Error('Unspent is empty');
+  }
+  const outputs = []
+  Object.keys(aggregatedOuts).forEach(color => {
+
+    // sum total unspent amount of one color from inputs
+    const inInputs = u => {
+      return inputs.findIndex(input =>
+        u.outpoint.hash.equals(input.prevout.hash) &&
+        (u.outpoint.index === input.prevout.index)) > -1;
+    }
+
+    const isColor = u => color === String(u.output.color);
+    let sum = unspent.filter(isColor).filter(inInputs).reduce((a, u) => add(a, BigInt(u.output.value)), BigInt(0));
+
+    Object.keys(aggregatedOuts[color]).forEach(from => {
+      aggregatedOuts[color][from].forEach(out => {
+        if (lessThan(sum, BigInt(out.amount))) {
+          throw new Error('Not enough inputs');
+        }
+        sum = subtract(sum, BigInt(out.amount));
+        outputs.push(new Output(out.amount, out.to.toLowerCase(), color));
+      })
+      // generate refund output for input
+      if (greaterThan(sum, BigInt(0))) {
+        outputs.push(new Output(sum, from.toLowerCase(), color));
+      }
+    })
+  })
+  return outputs;
+}
+
+// user should provide erc20 addresses used in tx's spend condition contracts,  
+module.exports = async (tx, erc20sToBeUsed, state) => {
+  if (tx.type !== Type.SPEND_COND) {
+    throw new Error('Spending Condition tx expected');
+  }
+
+  const colorAddrMap = {}
+  Object.keys(erc20sToBeUsed).forEach(item => {
+    colorAddrMap[item] = Buffer.from(erc20sToBeUsed[item].address.replace('0x', ''), 'hex')
+  })
+
+  const inputMap = {};
+  for (let i = 0; i < tx.inputs.length; i += 1) {
+    inputMap[tx.inputs[i].prevout.getUtxoId()] =
+      state.unspent[tx.inputs[i].prevout.hex()];
+  }
+
+  // creating a new VM instance
+  const vm = new VM({
+    hardfork: 'petersburg'
+  });
+
+  // creating the reactor account with some wei
+  const reactorAccount = new Account();
+  reactorAccount.balance = '0xf00000000000000001';
+  await setAccount(reactorAccount, REACTOR_ADDR, vm.stateManager);
+
+  let nonceCounter = 0;
+
+  // deploy erc20 contracts used in this simulation
+  // eslint-disable-next-line  guard-for-in
+  for (const color in Object.keys(colorAddrMap)) {
+    // eslint-disable-next-line no-await-in-loop
+    await setAccountCode(erc20sToBeUsed[color].erc20code || minErc20Code, colorAddrMap[color], vm.stateManager); // eslint-disable-line no-await-in-loop
```

Comment:
We should the use the ERC20 code from ERC20Min, if not we might get different results.